simba/optimization.py

```diff
@@ -276,107 +276,101 @@ def recombination(schedule, args, trips, depot_trips):
-                        f'{depot_trip["distance"] / 1000} km)')
+                        f'({trip_dict["travel_time"].total_seconds() / 60} min, '
+                        f'{trip_dict["distance"] / 1000} km)')
 
             # can trip be completed while having enough energy to reach depot again?
             # probe return trip
+            height_difference = schedule.get_height_difference(trip.arrival_name, depot_name)
             depot_trip = generate_depot_trip_data_dict(
                 trip.arrival_name, depot_name, depot_trips,
                 args.default_depot_distance, args.default_mean_speed)
-            height_difference = schedule.get_height_difference(trip.arrival_name,
-                                                               depot_name)
-            depot_trip = {
-                "departure_time": trip.arrival_time,
-                "departure_name": trip.arrival_name,
-                "arrival_time": trip.arrival_time + depot_trip["travel_time"],
-                "arrival_name": depot_name,
-                "distance": depot_trip["distance"],
-                "line": trip.line,
-                "charging_type": charging_type,
-                "temperature": trip.temperature,
-                "height_difference": height_difference,
-                "level_of_loading": 0,
-                "mean_speed": depot_trip["mean_speed"],
-                "station_data": schedule.station_data,
-            }
-            # rotation.add_trip needs dict, but consumption calculation is better done on Trip obj:
-            # create temporary depot trip object for consumption calculation
-            tmp_trip = Trip(rotation, **depot_trip)
-            # Sets tmp_trip.delta_soc and tmp_trip.consumption
-            schedule.calculate_trip_consumption(tmp_trip)
-            if soc >= -(trip.delta_soc + tmp_trip.delta_soc):
-                # next trip is possible: add trip, use info from original trip
-                trip_dict = vars(trip)
-                del trip_dict["rotation"]
-                trip_dict["charging_type"] = charging_type
-                rotation.add_trip(trip_dict)
+            depot_trip = Trip(
+                rotation=rotation,
+                departure_time=trip.arrival_time,
+                departure_name=trip.arrival_name,
+                arrival_time=trip.arrival_time + depot_trip["travel_time"],
+                arrival_name=depot_name,
+                distance=depot_trip["distance"],
+                line=trip.line,
+                temperature=trip.temperature,
+                height_difference=height_difference,
+                level_of_loading=0,
+                mean_speed=depot_trip["mean_speed"],
+            )
+
+            # check: can next trip be done with return to depot?
+            # create temp rotation (clone) to facilitate rollback
+            tmp_rot = deepcopy(rotation)
+            tmp_rot.add_trip(trip)
```

Comment:
can this be done differently? rotation keeping a reference to schedule means every deepcopy also deepcopies the whole schedule. Maybe generate a rotation explicitly via function or set schedule None beforehand?